src/main/java/jenkins/metrics/util/AutoSamplingHistogram.java

```diff
@@ -59,11 +67,23 @@ public AutoSamplingHistogram(Gauge<? extends Number> source, Reservoir reservoir
     }
 
     public void update() {
-        Number value = source.getValue();
-        if (value instanceof Integer) {
-            update(value.intValue());
-        } else {
-            update(value.longValue());
+        try {
+            Number value = source.getValue();
+            if (value instanceof Integer) {
+                update(value.intValue());
+            } else if (value != null) {
+                update(value.longValue());
+            } else {
+                LOGGER.log(Level.FINE, "Gauge {0} returned null", source);
```

Comment:
this is incorrect - as  source is specified as `? extends number`
so this could be a `Double` or `Float`